velox/exec/RowsStreamingWindowBuild.cpp

```diff
@@ -101,13 +122,17 @@ void RowsStreamingWindowBuild::noMoreInput() {
 }
 
 std::shared_ptr<WindowPartition> RowsStreamingWindowBuild::nextPartition() {
-  VELOX_CHECK(hasNextPartition());
-  return windowPartitions_[++outputPartition_];
+  return outputPartition();
 }
 
 bool RowsStreamingWindowBuild::hasNextPartition() {
-  return !windowPartitions_.empty() &&
-      outputPartition_ + 2 <= windowPartitions_.size();
+  // Remove the processed output partition from the queue.
+  while (!windowPartitions_.empty() && outputPartition()->complete() &&
```

Comment:
Does this need to be a while loop ? Partitions always have rows on construction.